src/main/java/net/sandrohc/schematic4j/builder/SchematicBuilder.java

```diff
@@ -0,0 +1,252 @@
+        this.width = width;
+        this.height = height;
+        this.length = length;
+        return (R) this;
+    }
+
+    /**
+     * Sets the schematic width (X axis).
+     */
+    public final <R extends SchematicBuilder> R width(int width) {
+        this.width = width;
+        return (R) this;
+    }
+
+    /**
+     * Sets the schematic height (Y axis).
+     */
+    public final <R extends SchematicBuilder> R height(int height) {
+        this.height = height;
+        return (R) this;
+    }
+
+    /**
+     * Sets the schematic length (Z axis).
+     */
+    public final <R extends SchematicBuilder> R length(int length) {
+        this.length = length;
+        return (R) this;
+    }
+
+    /**
+     * Sets the schematic's offset.
+     */
+    public SchematicBuilder offset(int x, int y, int z) {
+        this.offset = new SchematicBlockPos(x, y, z);
+        return this;
+    }
+
+    /**
+     * Sets the schematic's offset.
+     */
+    public SchematicBuilder offset(SchematicBlockPos pos) {
+        this.offset = pos;
+        return this;
+    }
+
+    /**
+     * Sets a block at a specific location.
+     * @param block A block string with states, for example {@code minecraft:grass_block[snowy=true]}
+     */
+    public SchematicNamedBuilder<SchematicBlock> block(String block) {
+        return new SchematicNamedBuilder<>(new SchematicBlock(block), (sblock, pos) -> {
+            blocks[pos.x][pos.y][pos.z] = sblock;
+            return this;
+        });
+    }
+
+    /**
+     * Defines the blocks inside the schematic.
+     */
+    public <R extends SchematicBuilder> R blocks(SchematicBlock[][][] blocks) {
+        this.blocks = blocks;
+        return (R) this;
+    }
+
+    /**
+     * Sets a block entity at a specific location.
+     * @param block The block type, for example {@code minecraft:chest}
+     * @param nbt The NBT data
+     */
+    public SchematicNamedBuilder<SchematicBlockEntity> blockEntity(String block, Map<String, Object> nbt) {
+        return new SchematicNamedBuilder<>(new SchematicBlockEntity(block, null, nbt), (blockEntity, pos) -> {
+            blockEntity.pos = pos;
+            blockEntities.add(blockEntity);
+            return this;
+        });
+    }
+
+    /**
+     * Adds an entity to a specific location.
+     * @param entity The entity type, for example {@code minecraft:creeper}
+     */
+    public SchematicEntityBuilder entity(String entity) {
+        return new SchematicEntityBuilder(new SchematicEntity(entity, null, null), (sentity, pos) -> {
+            sentity.pos = pos;
+            entities.add(sentity);
+            return this;
+        });
+    }
+
+    /**
+     * Exports the schematic into a {@link Schematic} object.
+     */
+    public abstract Schematic build();
+
+    /**
+     * Exports the schematic to a file.
+     * @param path The file path.
+     */
+    public abstract void exportToFile(String path);
```

Comment:
Probably better to leave the export functionality separate, so it can be used with any ordinary schematic. Something that can be used like this:
```java
Schematic schematic = SchematicBuilder
    .create(SchematicFormat.LITEMATICA)
    .block("minecraft:grass").at(0, 0, 0)
    .build();
SchematicExporter exporter = SchematicExporter.from(schematic); // Or schematic.export()
byte[] bytes = exporter.toBytes();
```

Reply:
Done